### PR TITLE
GRPC Interface for svc-storage

### PR DESCRIPTION
**Pull request by romanmandryk**
- GRPC Interface for svc-storage - first version with sample queries and inserts/updates
- in-memory db for e2e testing with other services (e.g. scheduler inserting, updating and reading flight plans)

### CODE REVIEW

File: .cspell.project-words.txt [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+protobuf
+repr
+RUSTC
+Rustup
+setaf
+SMUL
+smul
+taplo
+tput
+Vertiport
```
Comment (ServiceDog) [inline .cspell.project-words.txt:23]: I don't think it's case sensitive, for future reference

File: README.md [hunk 2]
```diff
@@ -13,7 +13,7 @@ Checks](https://github.com/arrow-air/svc-template-rust/actions/workflows/rust_ci
 Discord](https://img.shields.io/discord/853833144037277726?style=plastic)
 
 2. *Rename `svc-template-rust` and `svc_template_rust` in all files*
```
Comment (ServiceDog) [inline README.md:15]: Follow steps and remove if done

File: README.md [hunk 3]
```diff
@@ -41,8 +41,8 @@ cargo test
 
 ## :scroll: Documentation
 The following documents are relevant to this service:
-- [Concept of Operations](TODO)
-- [Requirements & User Stories](TODO)
+- [Concept of Operations](./docs/conops.md)
+- [Requirements & User Stories](./docs/requirements.md)
 - [SDD](./docs/sdd.md)
```
Comment (ServiceDog) [inline README.md:46]: add ICD link

File: client/examples/client.rs [hunk 4]
```diff
@@ -0,0 +1,44 @@
+//! gRPC client implementation
+
+///module svc_storage generated from svc-storage.proto
+pub mod svc_storage {
```
Comment (ServiceDog) [inline client/examples/client.rs:4]: Try using "use svc_storage_client::{etc.}"

File: client/src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,7 @@
+//! gRPC client library exposing svc_storage module
+
+///module svc_storage generated from svc-storage.proto
+pub mod svc_storage {
+    #![allow(unused_qualifications, missing_docs, missing_copy_implementations)]
```
Comment (ServiceDog) [inline client/src/lib.rs:5]: Can you add the copy implementations in the build.rs?

File: server/src/main.rs [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+    ) -> Result<Response<ReadyResponse>, Status> {
+        let response = ReadyResponse { ready: true };
+        Ok(Response::new(response))
+    }
+}
+
+///Main entry point: starts gRPC Server on specified address and port
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    dotenv().ok();
```
Comment (ServiceDog) [inline server/src/main.rs:183]: Add a comment for what this is doing

**Review by ServiceDog — CHANGES_REQUESTED** [review]

File: server/src/main.rs (hunk 6 above)
Comment (romanmandryk) [reply server/src/main.rs:184]: Did you mean  what `dotenv().ok();` is doing? I added comment for that.

File: README.md [hunk 7]
```diff
@@ -1,28 +1,12 @@
 ![Arrow Banner](https://github.com/Arrow-air/.github/raw/main/profile/assets/arrow_v2_twitter-banner_neu.png)
 
-# MODULE_NAME Service
-
-*TODO after cloning:*
-
-1. *Replace the repository name for each:*
-
-![Rust
```
Comment (ServiceDog) [inline README.md:9]: Keep the badges, just replace the svc-template-rust part

File: README.md [hunk 8]
```diff
@@ -1,28 +1,12 @@
...
-
-2. *Rename `svc-template-rust` and `svc_template_rust` in all files*
-   - *Replace with the name of your service (e.g. `svc-scheduler`)*
-3. *Rename or remove `tmp_lib`*
-4. *Remove this and previous numbered bullets*
-
-
+# svc-storage Service
 ## :telescope: Overview
 *TODO: This is a high level description of this module.*
```
Comment (ServiceDog) [inline README.md:5]: Short description of the module

**Review by ServiceDog — CHANGES_REQUESTED** [review]

File: README.md (hunk 7 above)
Comment (romanmandryk) [reply README.md:9]: Ok, got it now. I think I removed badges on svc-scheduler too

**Review by ServiceDog — APPROVED** [review]

**Pull request merged**